slycot/transform.py

```diff
@@ -1072,132 +1093,141 @@ def tg01fd(l,n,m,p,A,E,B,C,Q=None,Z=None,compq='N',compz='N',joba='N',tol=0.0,ld
+        = 'N':  do not compute Q.
+        = 'I':  Q is initialized to the unit matrix, and the
+                orthogonal matrix Q is returned.
+        = 'U':  Q must contain an orthogonal matrix Q1 on entry,
+                and the product Q1*Q is returned.
+        Default is 'N'.
+    compz :  {'N', 'I', 'U'}, optional
+        = 'N':  do not compute Z.
+        = 'I':  Z is initialized to the unit matrix, and the
+                orthogonal matrix Z is returned.
+        = 'U':  Z must contain an orthogonal matrix Z1 on entry,
+                and the product Z1*Z is returned.
+        Default is 'N'.
+    joba :  {'N', 'R', 'T'}, optional
+        = 'N':  do not reduce A22.
+        = 'R':  reduce A22 to a SVD-like upper triangular form.
+        = 'T':  reduce A22 to an upper trapezoidal form.
+        Default is 'N'.
+    tol : float, optional
+        The tolerance to be used in determining the rank of E
+        and of A22. If the user sets TOL > 0, then the given
+        value of TOL is used as a lower bound for the
+        reciprocal condition numbers of leading submatrices
+        of R or R22 in the QR decompositions E * P = Q * R of E
+        or A22 * P22 = Q22 * R22 of A22.
+        A submatrix whose estimated condition number is less than
+        1/TOL is considered to be of full rank.  If the user sets
+        TOL <= 0, then an implicitly computed, default tolerance,
+        defined by  TOLDEF = L*N*EPS,  is used instead, where
+        EPS is the machine precision (see LAPACK Library routine
+        DLAMCH). TOL < 1.
+        Default is `0.0`.
+    ldwork : int, optional
+        The length of the cache array.
+        ldwork >= MAX( 1, n+p, MIN(l,n)+MAX(3*n-1,m,l) ).
+        For optimal performance, ldwork should be larger.
+        Default is None.
+    
+    Returns
+    -------
+    A : (l, n) ndarray
+        On entry, the leading L-by-N part of this array must
+        contain the state dynamics matrix A.
+        On exit, the leading L-by-N part of this array contains
+        the transformed matrix Q'*A*Z. If JOBA = 'T', this matrix
+        is in the form
+
+                        ( A11  *   *  )
+            Q'*A*Z = (  *   Ar  X  ) ,
+                        (  *   0   0  )
+
+        where A11 is a RANKE-by-RANKE matrix and Ar is a
+        RNKA22-by-RNKA22 invertible upper triangular matrix.
+        If JOBA = 'R' then A has the above form with X = 0.
+    E : (l, n) ndarray
+        The leading L-by-N part of this array contains
+        the transformed matrix Q'*E*Z.
+
+                    ( Er  0 )
+        Q'*E*Z = (       ) ,
+                    (  0  0 )
+
+        where Er is a RANKE-by-RANKE upper triangular invertible
+        matrix.
+    B : (l, m) ndarray
+        The leading L-by-M part of this array contains
+        the transformed matrix Q'*B.
+    C : (p, n) ndarray
+        The leading P-by-N part of this array contains
+        the transformed matrix C*Z.
+    Q : (l, l) ndarray
```

Comment:
Please bring the order into the same one as the return tuple:

```
return A,E,B,C,ranke,rnka22,Q,Z
```

Reply:
Done